src/mongo/jobs/EnsureIndexes.class.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Tripod\Mongo\Jobs;
+
+/**
+ * Class EnsureIndexes
+ * @package Tripod\Mongo\Jobs
+ */
+class EnsureIndexes extends JobBase {
+
+    const STORENAME_KEY  = 'storeName';
+    const REINDEX_KEY    = 'reindex';
+
+    /**
+     * Runs the EnsureIndexes Job
+     * @throws \Exception
+     */
+    public function perform()
+    {
+        try
+        {
+            $this->debugLog("[JOBID " . $this->job->payload['id'] . "] EnsureIndexes::perform() start");
+
+            $timer = new \Tripod\Timer();
+            $timer->start();
+
+            $this->validateArgs();
+
+            \Tripod\Mongo\Config::setConfig($this->args[self::TRIPOD_CONFIG_KEY]);
+
+            $this->getIndexUtils()->ensureIndexes(
+                $this->args[self::REINDEX_KEY],
+                $this->args[self::STORENAME_KEY],
+                true // always create indexes in the background
+            );
+
+            $timer->stop();
+            // stat time taken to process job, from time it was picked up
+            $this->getStat()->timer(MONGO_QUEUE_ENSURE_INDEXES_SUCCESS,$timer->result());
+            $this->debugLog("[JOBID " . $this->job->payload['id'] . "] EnsureIndexes::perform() done in {$timer->result()}ms");
+        }
+        catch(\Exception $e)
+        {
+            $this->getStat()->increment(MONGO_QUEUE_ENSURE_INDEXES_FAIL);
+            $this->errorLog("Caught exception in ".get_class($this).": ".$e->getMessage());
+            throw $e;
+        }
+    }
+
+    /**
+     * @param string $storeName
```

Comment:
Can we get a docblock here to explain that the function here is the one that creates the job.